compiler/hash-typecheck/src/traverse/mod.rs

```diff
@@ -1488,44 +1512,157 @@ impl<'gs, 'ls, 'cd, 'src> visitor::AstVisitor for TcVisitor<'gs, 'ls, 'cd, 'src>
         todo!()
     }
 
-    type StructDefEntryRet = TermId;
+    type StructDefEntryRet = Param;
 
     fn visit_struct_def_entry(
         &mut self,
-        _ctx: &Self::Ctx,
-        _node: hash_ast::ast::AstNodeRef<hash_ast::ast::StructDefEntry>,
+        ctx: &Self::Ctx,
+        node: hash_ast::ast::AstNodeRef<hash_ast::ast::StructDefEntry>,
     ) -> Result<Self::StructDefEntryRet, Self::Error> {
-        todo!()
+        let walk::StructDefEntry { name, ty, default } =
+            walk::walk_struct_def_entry(self, ctx, node)?;
+
+        // Try and figure out a known term...
+        let (ty, default_value) = match (ty, default) {
```

Comment:
Maybe add a @@Repetition to factor this into a function since it is also used for declarations and tuples.